### PR TITLE
MDC button trailing icon support

### PR DESCRIPTION
**Pull request by gugu**
Now MDC-web supports trailing icon

https://github.com/material-components/material-components-web/tree/master/packages/mdc-button#trailing-icon

### CODE REVIEW

File: packages/button/index.tsx [hunk 1]
```diff
@@ -37,6 +37,7 @@ export interface ButtonProps<T extends ButtonTypes> extends Ripple.InjectedProps
   className?: string;
   icon?: React.ReactElement<React.HTMLProps<HTMLOrSVGElement>>;
   href?: string;
+  trailingIcon?: boolean,
```
Comment (moog16) [inline packages/button/index.tsx]: what if instead you changed this to accept

```
trailingIcon: React.ReactElement<React.HTMLProps<HTMLOrSVGElement>>;
```

File: packages/button/index.tsx [hunk 2]
```diff
@@ -74,16 +76,22 @@ export const Button = <T extends ButtonTypes>(
   if (href) {
     return (
       <a {...props as React.HTMLProps<HTMLAnchorElement>} href={href}>
-        {renderIcon(icon)}
-        {children}
+        {!trailingIcon ? renderIcon(icon) : null}
+        <span className="mdc-button__label">
+          {children}
+        </span>
+        {trailingIcon ? renderIcon(icon) : null}
```
Comment (moog16) [inline packages/button/index.tsx]: it could then be renderIcon(trailingIcon)...does that make sense?

File: packages/button/package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "url": "https://github.com/material-components/material-components-web-react.git"
   },
   "dependencies": {
-    "@material/button": "^0.41.0",
+    "@material/button": "^0.43.0",
```
Comment (moog16) [inline packages/button/package.json]: this also needs to be updated in the main `./package.json`

**Comment by gugu** [comment]
I made these changes and added some simple test. This morning I'll check if there are more tests needed. If yes - I will add them. And retest module in our application

File: packages/button/index.tsx (hunk 2 above)
Comment (moog16) [reply packages/button/index.tsx]: @gugu I meant for the type change to change this line into 
```
{renderIcon(trailingIcon)}
```

File: test/unit/button/index.test.tsx [hunk 4]
```diff
@@ -24,6 +24,12 @@ test('renders an icon', () => {
   assert.isTrue(wrapper.find('.test-icon').hasClass('mdc-button__icon'));
 });
 
+test('renders a trailing icon', () => {
```
Comment (moog16) [inline test/unit/button/index.test.tsx]: could you also add a test for when it is an anchor tag?

**Comment by moog16** [comment]
Left some comments, but I think once those are addressed it should be enough.

**Comment by gugu** [comment]
Done

**Comment by moog16** [comment]
@gugu please update button golden to the same as this: https://github.com/material-components/material-components-web-react/pull/625

**Comment by moog16** [comment]
https://github.com/material-components/material-components-web-react/pull/625 tests passing...looks good!

**Pull request merged**